src/anyio/_core/_fileio.py

```diff
@@ -129,22 +129,22 @@ async def flush(self) -> None:
 async def open_file(file: Union[str, PathLike, int], mode: OpenBinaryMode,
                     buffering: int = ..., encoding: Optional[str] = ...,
                     errors: Optional[str] = ..., newline: Optional[str] = ..., closefd: bool = ...,
-                    opener: Optional[Callable] = ...) -> AsyncFile[bytes]:
+                    opener: Optional[Callable[[str, int], int]] = ...) -> AsyncFile[bytes]:
```

Comment:
```suggestion
                    opener: Optional[Callable[Union[str, int], int]] = ...) -> AsyncFile[bytes]:
```

https://github.com/python/typeshed/blob/0085539df30b59d53e94980faddbf695800584a7/stdlib/io.pyi#L74